### ISSUE TITLE
Image_Preloader with mode=folder bug? Label mis-assigned.  Problem Solved.

### ISSUE THREAD

**Issue by hyy1111** [tflearn/tflearn#506]
I used the image_preloader(mode=folder) to load my image data and trained on inception_v3 network, the train accuracy and validation(I prepared test sets) accuracy arrived at 98% and 90% respectively, but when I  used model.predict() on some train images and test images, it all output same and wrong labels. For example, when input images in train folder 49 and test folder 49, the model.predict() outputs 44 for both. Anybody know why the labels are mis-assigned? 

Code:
# Load path/class_id image file:
train_dataset_folder = '/home/cclabs/PycharmProjects/test_tflearn_3D/train_images/Depth'
test_dataset_folder = '/home/cclabs/PycharmProjects/test_tflearn_3D/test_images/Depth'

X, Y = image_preloader(train_dataset_folder, image_shape=(227, 227),   mode='folder', categorical_labels=True,   normalize=True)
X_test, Y_test = image_preloader(test_dataset_folder, image_shape=(227, 227),   mode='folder', categorical_labels=True,   normalize=True)

inceptionv3 network
...
...
...

model = tflearn.DNN(network, best_checkpoint_path='best_checkpoint_path_3D_1/model_incep_v3_',
                    max_checkpoints=1, tensorboard_verbose=2)

model.fit(X, Y, n_epoch=100, validation_set=(X_test, Y_test), shuffle=True,
          show_metric=True, batch_size=64, snapshot_step=100,
          snapshot_epoch=True)

**Comment by cristi-zz** [comment]
Maybe something to do with alphabetical ordering of the folders? Can you try to generate a txt file as input for image_preloader? This way, you control the label coding.

**Comment by hyy1111** [comment]
Thanks for your reply cristi-zz, I found  image_preloader with mode=folder has bug. I organized the image folders as in http://tflearn.org/data_utils/, there are 50 folders named 0 to 49. And when I plot the last image (X[-1]), It showed a image in folder 9, but the label is for 49. I think the image_preloader with mode=folder does not load images in folder names order (0 to 49)
